migrate/starter/starter.go

```diff
@@ -61,6 +62,11 @@ func checkStartVersion(args []string) version {
 	if err != nil {
 		return internalV2
 	}
+	if fs.Lookup("version").Value.String() == "true" {
+		fmt.Println("etcd version:", etcdversion.Version, "(allow legacy mode)")
```

Comment:
it is in a mixed state, which can use v2 in the following, or exec v0.4 if it doesn't upgrade. I have concerns that users have false thoughts that etcd v2.0 can run v0.4 data dir directly.